src/qilib/utils/storage/mongo.py

```diff
@@ -71,12 +73,17 @@ def __init__(self, name: str, host: str = 'localhost', port: int = 27017, databa
             host: MongoDB host
             port: MongoDB port
             database: The database to use, if empty the name of the storage is used
+            connection_timeout: How long to try to connect to database before raising an error
+        Raises:
+            StorageTimeoutError: If connection to database has not been established before connection_timeout is reached
         """
         super().__init__(name)
 
         type_registry = TypeRegistry([NumpyArrayCodec()])
         codec_options = CodecOptions(type_registry=type_registry)
-        self._client = MongoClient(host, port)
+        self._client = MongoClient(host, port, serverSelectionTimeoutMS=connection_timeout)
+        self._check_server_connection(connection_timeout)
+        self._client.server_info()
```

Comment:
Is this line needed,  as part of function call in line 85, it would be executed.

Reply:
No, not needed and has no been removed